src/main/java/eu/bbmri_eric/negotiator/mappers/NegotiationModelAssembler.java

```diff
@@ -95,40 +95,30 @@ private List<Link> getLinks(Page<NegotiationDTO> page, NegotiationState status)
   }
 
   private List<Link> getLinks(
-      Page<NegotiationDTO> page, NegotiationRole negotiationRole, Long userId) {
+      Page<NegotiationDTO> page, NegotiationRequestParameters filters, Long userId) {
     List<Link> links = new ArrayList<>();
     if (page.hasPrevious()) {
       links.add(
-          linkTo(
-                  methodOn(NegotiationController.class)
-                      .listRelated(userId, negotiationRole, page.getNumber() - 1, page.getSize()))
+          linkTo(methodOn(NegotiationController.class).listRelated(userId, filters))
```

Comment:
definitely not easily :D,  had to refactor it and get rid of the NegotiatorRequestParams because it's not handled by spring hateoas https://github.com/spring-projects/spring-hateoas/issues/1815. Had to be done